common/src/main/java/com/wynntils/core/chat/ChatManager.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright © Wynntils 2022.
+ * This file is released under AGPLv3. See LICENSE for full license details.
+ */
+package com.wynntils.core.chat;
+
+import com.wynntils.core.WynntilsMod;
+import com.wynntils.mc.event.ChatReceivedEvent;
+import com.wynntils.wc.utils.WynnUtils;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.minecraft.network.chat.Component;
+import net.minecraftforge.eventbus.api.EventPriority;
+import net.minecraftforge.eventbus.api.SubscribeEvent;
+
+public class ChatManager {
+    private static final Pattern INFO_BAR_PATTERN = Pattern.compile("§c❤ (\\d+)\\/(\\d+)§0 .* §b✺ (\\d+)\\/(\\d+)");
+
+    public static void init() {
+        WynntilsMod.getEventBus().register(ChatManager.class);
+    }
+
+    @SubscribeEvent(priority = EventPriority.HIGHEST)
+    public static void onChatReceived(ChatReceivedEvent e) {
+        if (!WynnUtils.onWorld()) return;
+
+        switch (e.getType()) {
+            case GAME_INFO -> parseGameInfo(e.getMessage());
+            case SYSTEM -> parseSystemMessage(e.getMessage());
+            case CHAT -> parseChatMessage(e.getMessage());
```

Comment:
Perhaps a less universal name? (although I cant suggest a better one)

Reply:
Which is too universal you mean? `getType()`, or any of the enum values, or `parseChatMessage()`?

Reply:
CHAT.